core/src/main/java/org/elasticsearch/search/rescore/RescorerBuilder.java

```diff
@@ -124,12 +121,53 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     protected abstract void doXContent(XContentBuilder builder, Params params) throws IOException;
 
-    public abstract QueryRescoreContext build(QueryShardContext context) throws IOException;
+    /**
+     * Build the {@linkplain RescoreContext} that will be used to actually
+     * execute the rescore against a particular shard.
+     */
+    public final RescoreContext buildContext(QueryShardContext context) throws IOException {
+        RescoreContext rescoreContext = innerBuildContext(new RescoreContextSupport() {
```

Comment:
I'm ok with passing in `QueryShardContext` but it still looks like it exposes a lot of stuff! Cluster alias and the bitsetFilterCache and stuff. But, yeah, queries have access to all of it, so why not rescore?